ansible_collections/arista/avd/plugins/module_utils/eos_designs_facts/topology.py

```
@@ -0,0 +1,332 @@
+        Instance of Ansible 'template' lookup module used to render ip addresses with user-defined templates
+
+    Returns
+    -------
+    dict
+        topology.links: List of uplinks with all parameters
+        topology.peers: List of all uplink peers
+    """
+
+    switch = vars.get('switch', {})
+
+    if switch.get('uplink_type') == 'p2p':
+        topology = get_topology_p2p(vars, template_lookup_module)
+
+    elif switch.get('uplink_type') == 'port-channel':
+        topology = get_topology_port_channel(vars, filters)
+
+    # Special cases of setting ips during topology facts run,
+    # since these functions relies on data set in avd_switch_facts
+    if switch.get('vtep') is True:
+        switch['vtep_ip'] = get_vtep_ip(vars, template_lookup_module)
+
+    if switch.get('mlag') is True:
+        switch.update(get_mlag_ips(vars, template_lookup_module))
+
+    sanitized_topology = {}
+    for key in topology:
+        if topology[key] is not None:
+            sanitized_topology[key] = topology[key]
+
+    return {"topology": sanitized_topology, "switch": switch}
+
+
+def get_topology_p2p(vars, template_lookup_module):
+    """
+    Get topology.* facts for device with routed p2p uplinks
+
+    Parameters
+    ----------
+    vars : dict
+        Hostvars for one host
+    template_lookup_module: func
+        Instance of Ansible 'template' lookup module used to render ip addresses with user-defined templates
+
+    Returns
+    -------
+    dict
+        topology.links: List of uplinks with all parameters
+        topology.peers: List of all uplink peers
+    """
+
+    switch = vars['switch']
+    topology = vars.get('topology', {})
+
+    topology['links'] = []
+    topology['peers'] = []
+
+    uplink_interfaces = switch.get('uplink_interfaces', [])
+
+    for uplink_index, uplink_interface in enumerate(uplink_interfaces):
+        uplink_switches = switch.get('uplink_switches', [])
+        if len(uplink_switches) > uplink_index:
```

so what happens if there are more `uplink_switches` than uplink_interfaces`?  Looks like we would just ignore the end of the list but maybe worth outputting some warning if we can

---

We try to forgive missing entries in either list here. So just silently ignoring the missing entries. This is similar to what we have in Jinja today.
We have some incoming change which will set some of these values based on the platform. With that change, we will need the `uplink_switches` to be the "main" iterator, and then just ignore if there are more interfaces available. Only if we have fewer interfaces, we should eject a warning.